yarn-project/circuits.js/src/abis/abis.ts

```diff
@@ -359,9 +360,40 @@ export function computeTxHash(wasm: IWasmModule, txRequest: TxRequest): Fr {
  * @param callStackItem - The call stack item.
  * @returns The call stack item hash.
  */
-export function computeCallStackItemHash(wasm: IWasmModule, callStackItem: PublicCallStackItem): Fr {
+export function computeCallStackItemHash(
+  wasm: IWasmModule,
+  callStackItem: PrivateCallStackItem | PublicCallStackItem,
+): Fr {
+  if (callStackItem instanceof PrivateCallStackItem) {
+    return computePrivateCallStackItemHash(wasm, callStackItem);
+  } else if (callStackItem instanceof PublicCallStackItem) {
+    return computePublicCallStackItemHash(wasm, callStackItem);
+  } else {
+    throw new Error(`Unexpected call stack item type`);
+  }
+}
+
+/**
+ * Computes a call stack item hash.
+ * @param wasm - Relevant WASM wrapper.
+ * @param callStackItem - The call stack item.
+ * @returns The call stack item hash.
+ */
+export function computePrivateCallStackItemHash(wasm: IWasmModule, callStackItem: PrivateCallStackItem): Fr {
+  wasm.call('pedersen__init');
```

Comment:
Aside: I wonder if this is what's slowing down our tests.

Reply:
I understand it only runs once per process:

https://github.com/AztecProtocol/aztec-packages/blob/c2b18c1c12ccdc0ab26d8fb0ca83f5532b8e44bc/circuits/cpp/barretenberg/cpp/src/barretenberg/crypto/generators/generator_data.cpp#L215-L217